src/utils/settings/splitFilters.js

```diff
@@ -0,0 +1,111 @@
+ * Exposed for testing purposes.
+ * The possible formats of the query string are:
+ *  - '': if both byNameList and byPrefixList are undefined
+ *  - 'names=<comma-separated-values>': if only byPrefixList is undefined
+ *  - 'prefixes=<comma-separated-values>': if only byNameList is undefined
+ *  - 'names=<comma-separated-values>&prefixes=<comma-separated-values>': if no one is undefined
+ *
+ * @param {string[] | undefined} byNameList undefined or list of ordered and unique non-empty strings
+ * @param {string[] | undefined} byPrefixList undefined or list of ordered and unique non-empty strings
+ * @returns undefined or string with the `query` component of URL.
+ */
+function queryStringBuilder(byNameList, byPrefixList) {
+  const queryParams = [];
+  if (byNameList && byNameList.length > 0) queryParams.push('names=' + byNameList.map(value => encodeURIComponent(value)).join(','));
+  if (byPrefixList && byPrefixList.length > 0) queryParams.push('prefixes=' + byPrefixList.map(value => encodeURIComponent(value)).join(','));
+  return queryParams.length > 0 ? queryParams.join('&') : undefined;
+}
+
+/**
+ * Validates `splitFilters` configuration object and parses it to build the query string for filtering splits in `/splitChanges` fetch.
+ *
+ * @param {Object} settings factory configuration object
+ * @returns undefined or sanitized `splitFilters` array object with parsed `queryString`.
+ * @throws Error if the saniti some of the filters exceed the max allowed length
+ */
+export function splitFiltersBuilder(settings) {
+  const { sync: { splitFilters }, mode } = settings;
+
+  // do nothing if `splitFilters` param is not a non-empty array or mode is not STANDALONE
+  if (!splitFilters) return;
+  if (mode !== STANDALONE_MODE) {
+    log.warn(`Split filters have been configured but will have no effect if mode is not '${STANDALONE_MODE}', since synchronization is being deferred to an external tool`);
+    return;
+  }
+  if (!Array.isArray(splitFilters) || splitFilters.length === 0) {
+    log.warn('splitFilters configuration must be a non-empty array of filters');
+    return;
+  }
+
+  // validate filters and group their values by filter type
+  const filters = {
+    byName: undefined,
+    byPrefix: undefined
+  };
+  const validFilters = splitFilters.filter((filter, index) => {
+    if (filter && isString(filter.type) && Array.isArray(filter.values)) {
+      if (Object.prototype.hasOwnProperty.call(filters, filter.type)) {
+        if (!filters[filter.type]) filters[filter.type] = [];
+        filters[filter.type] = filters[filter.type].concat(filter.values);
+        return true;
+      } else {
+        log.warn(`'${filter.type}' is an invalid filter. Only 'byName' and 'byPrefix' are valid.`);
+      }
+    } else {
+      log.warn(`Split filter at position '${index}' is invalid. It must be an object with a valid 'type' filter and a list of 'values'.`);
```

Comment:
Move to input validation too. Maybe instead of position, use the word "index" since that way everyone will know 1 is not the first element on the list.